api/base/filters.py

```diff
@@ -61,6 +66,83 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class NewDjangoFilterMixin(django_filters.FilterSet):
+
+    QUERY_PATTERN = re.compile(r'^filter\[(?P<fields>((?:,*\s*\w+)*))\](\[(?P<op>\w+)\])?$')
+    FILTER_FIELDS = re.compile(r'(?:,*\s*(\w+)+)')
+
+    or_fields = []
+
+    def __init__(self, data=None, *args, **kwargs):
+        if data:
+            new_qd = QueryDict('', mutable=True)
+            for key, value in data.iteritems():
+                match = self.QUERY_PATTERN.match(key)
+                if match:
+                    match_dict = match.groupdict()
+                    fields = match_dict['fields']
+                    field_names = re.findall(self.FILTER_FIELDS, fields.strip())
+
+                    self.or_fields = []
+                    multiple_values = value.split(',')
+                    if len(multiple_values) > 1 or len(field_names) > 1:
```

Comment:
Ah, I think I see. But don't filters with multiple values need to be handled separately from filters with multiple field names anyway? Also, doesn't django-filter have a built in way to handle multiple values? Intuitively, `?full_name=Erin,Steve` should be translated to `.filter(full_name__in=['Erin', 'Steve'])`; I wonder if that's supported out-of-the-box.